lib/src/patterns.dart

```diff
@@ -124,3 +124,73 @@ final tablePattern = RegExp(
 /// A pattern which should never be used. It just satisfies non-nullability of
 /// pattern fields.
 final dummyPattern = RegExp('');
+
+/// A [String] pattern to match a named tag like `<table>` or `</table>`.
+const _namedTagDefinition =
+    // Opening tag begins.
+    '<'
+
+    // Tag name.
+    '[a-z][a-z0-9-]*'
+
+    // Attribute begins, see
+    // https://spec.commonmark.org/0.30/#attribute.
+    r'(?:\s+'
+
+    // Attribute name, see
+    // https://spec.commonmark.org/0.30/#attribute-name.
+    '[a-z_:][a-z0-9._:-]*'
+
+    //
+    '(?:'
+    // Attribute value specification, see
+    // https://spec.commonmark.org/0.30/#attribute-value-specification.
+    r'\s*=\s*'
+
+    // Attribute value, see
+    // https://spec.commonmark.org/0.30/#unquoted-attribute-value.
+    r'''(?:[^\s"'=<>`]+?|'[^']*?'|"[^"]*?")'''
+
+    // Attribute ends.
+    ')?)*'
+
+    // Opening tag ends.
+    r'\s*/?>'
+
+    //
```

Comment:
Is the comment here just to preserve newlines? Could say something like 'OR'

Reply:
Fixed